src/common/raft/integration/raft_knowhere_index.hpp

```diff
@@ -18,52 +18,11 @@
 #include <cstdint>
 
 #include "common/raft/integration/raft_knowhere_config.hpp"
+#include "common/raft/integration/type_mappers.hpp"
 #include "common/raft/proto/raft_index_kind.hpp"
 
 namespace raft_knowhere {
 
-using knowhere_data_type = float;
```

Comment:
We move these to their own header because we need them in the extern template instantiations, and we do not want to have to recompile the extern templates every time we change the connector code between RAFT and knowhere.